pkg/nsx/cluster.go

```diff
@@ -460,3 +465,23 @@ func (nsxVersion *NsxVersion) featureSupported(feature int) bool {
 	}
 	return false
 }
+
+func (cluster *Cluster) getLicenseFromNsx() (*util.NsxLicense, error) {
+	resp, err := cluster.httpAction(LicenseAPI, "GET")
+	if err != nil {
+		log.Error(err, "failed to get nsx license")
+		return nil, err
+	}
+	nsxLicense := &util.NsxLicense{}
+	err, _ = util.HandleHTTPResponse(resp, nsxLicense, true)
+	return nsxLicense, err
+}
+
+func (cluster *Cluster) FetchLicense() error {
+	nsxLicense, err := cluster.getLicenseFromNsx()
+	if err != nil {
+		return err
+	}
+	util.UpdateFeatureLicense(nsxLicense)
+	return nil
+}
```

Comment:
getLicenseFromNsx could be put in FetchLicense, no need to have another function